common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -136,7 +136,11 @@
   private DB db;
 
   public YarnShuffleService() {
-    super("spark_shuffle");
+    this("spark_shuffle");
+  }
+
+  public YarnShuffleService(String serviceName) {
```

Comment:
Is it needed to add new constructor?

Reply:
This is mostly to be able to test it since `yarn.nodemanager.aux-services.spark_shuffle.class` option assumes no-args constructor. Maybe make it `protected`?